src/Workspaces/Core/Portable/Utilities/NullableHelpers/NamedTypeSymbolWithNullableAnnotation.cs

```diff
@@ -0,0 +1,70 @@
+{
+    internal static partial class NullableExtensions
+    {
+        private sealed class NamedTypeSymbolWithNullableAnnotation : TypeSymbolWithNullableAnnotation, INamedTypeSymbol
+        {
+            public NamedTypeSymbolWithNullableAnnotation(INamedTypeSymbol wrappedSymbol, NullableAnnotation nullability) : base(wrappedSymbol, nullability)
+            {
+            }
+
+            public override void Accept(SymbolVisitor visitor)
+            {
+                visitor.VisitNamedType(this);
+            }
+
+            public override TResult Accept<TResult>(SymbolVisitor<TResult> visitor)
+            {
+                return visitor.VisitNamedType(this);
+            }
+
+            #region INamedTypeSymbol Implementation Forwards
+
+            public int Arity => WrappedSymbol.Arity;
+            public bool IsGenericType => WrappedSymbol.IsGenericType;
+            public bool IsUnboundGenericType => WrappedSymbol.IsUnboundGenericType;
+            public bool IsScriptClass => WrappedSymbol.IsScriptClass;
+            public bool IsImplicitClass => WrappedSymbol.IsImplicitClass;
+            public bool IsComImport => WrappedSymbol.IsComImport;
+            public IEnumerable<string> MemberNames => WrappedSymbol.MemberNames;
+            public ImmutableArray<ITypeParameterSymbol> TypeParameters => WrappedSymbol.TypeParameters;
+            public ImmutableArray<ITypeSymbol> TypeArguments => WrappedSymbol.TypeArguments;
+            public ImmutableArray<NullableAnnotation> TypeArgumentsNullableAnnotations => WrappedSymbol.TypeArgumentsNullableAnnotations;
+            public IMethodSymbol DelegateInvokeMethod => WrappedSymbol.DelegateInvokeMethod;
+            public INamedTypeSymbol EnumUnderlyingType => WrappedSymbol.EnumUnderlyingType;
+            public INamedTypeSymbol ConstructedFrom => WrappedSymbol.ConstructedFrom;
+            public ImmutableArray<IMethodSymbol> InstanceConstructors => WrappedSymbol.InstanceConstructors;
+            public ImmutableArray<IMethodSymbol> StaticConstructors => WrappedSymbol.StaticConstructors;
+            public ImmutableArray<IMethodSymbol> Constructors => WrappedSymbol.Constructors;
+            public ISymbol AssociatedSymbol => WrappedSymbol.AssociatedSymbol;
+            public bool MightContainExtensionMethods => WrappedSymbol.MightContainExtensionMethods;
+            public INamedTypeSymbol TupleUnderlyingType => WrappedSymbol.TupleUnderlyingType;
+            public ImmutableArray<IFieldSymbol> TupleElements => WrappedSymbol.TupleElements;
+            public bool IsSerializable => WrappedSymbol.IsSerializable;
+            private new INamedTypeSymbol WrappedSymbol => (INamedTypeSymbol)base.WrappedSymbol;
+            INamedTypeSymbol INamedTypeSymbol.OriginalDefinition => WrappedSymbol.OriginalDefinition;
+
+            public INamedTypeSymbol Construct(params ITypeSymbol[] typeArguments)
+            {
+                return WrappedSymbol.Construct(typeArguments);
```

Comment:
is this correct?  say i have  `IList<T>?` and i construct that with `int`, shouldn't i get `IList<int>?`

Reply:
Good catch. These were just autogenerated with the "implement interface through" so clearly I get to audit them.